### ISSUE TITLE
[Bug]: The pkg tool can no longer remove or upgrade packages because of missing dependencies

### ISSUE THREAD

**Issue by Richard092345** [termux/termux-packages#19386]
### Problem description

The command 'pkg uninstall ...' does not remove the oackage but instead tells me that a dependency is missing in package Octave and it asks me to run 'apt --fix-broken install' to fix the issue. Unfortunately that fails and I end up being unable to either remove or upgrade any package.

### What steps will reproduce the bug?

$ pkg install octave
Checking availability of current mirror:
[*] https://mirrors.ustc.edu.cn/termux/apt/termux-main: ok
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
octave is already the newest version (8.4.0-2).
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 arpack-ng : Depends: libopenblas but it is not going to be installed
 octave : Depends: libopenblas but it is not going to be installed
          Depends: qrupdate-ng but it is not going to be installed
 suitesparse : Depends: libopenblas but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
~

### What is the expected behavior?

I would expect 'pkg remove ...' to run for any package even in the case where a dependency is missing. The dependency is missing because 'pkg' can't install it. 

### System information

termux-info:
```
Termux Variables:
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=24043
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://mirrors.ustc.edu.cn/termux/apt/termux-main stable main
# sources.list.d/pointless.list
deb https://its-pointless.github.io/files/21 termux extras
Updatable packages:
bash-completion/stable 2.12.0 all [upgradable from: 2.11-2]
command-not-found/stable 2.4.0-13 aarch64 [upgradable from: 2.4.0-11]
curl/stable 8.6.0-1 aarch64 [upgradable from: 8.6.0]
git/stable 2.44.0 aarch64 [upgradable from: 2.43.2]
libblkid/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
libcurl/stable 8.6.0-1 aarch64 [upgradable from: 8.6.0]
libdav1d/stable 1.4.0 aarch64 [upgradable from: 1.3.0]
libjasper/stable 4.2.1 aarch64 [upgradable from: 4.2.0]
libksba/stable 1.6.6 aarch64 [upgradable from: 1.6.5-1]
liblzma/stable 5.6.0-1 aarch64 [upgradable from: 5.4.6]
libmount/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
libpng/stable 1.6.43 aarch64 [upgradable from: 1.6.42]
libsmartcols/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
libuuid/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
mount-utils/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
mpg123/stable 1.32.5 aarch64 [upgradable from: 1.32.4]
openssl/stable 1:3.2.1-1 aarch64 [upgradable from: 1:3.2.1]
util-linux/stable 2.39.3-4 aarch64 [upgradable from: 2.39.3-3]
xz-utils/stable 5.6.0-1 aarch64 [upgradable from: 5.4.6]
termux-tools version:
1.40.6
Android version:
12
Kernel build information:
Linux localhost 5.10.81-android12-9-24790453-abX900XXS2AVJ4 #1 SMP PREEMPT Sun Oct 23 04:45:23 UTC 2022 aarch64 Android
Device manufacturer:
samsung
Device model:
SM-X900
LD Variables:
LD_LIBRARY_PATH=/data/data/com.termux/files/usr/lib/
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
```
```

```

**Comment by Richard092345** [comment]
I managed to remove octave with the following command. I will not reinstall it right now but wait until I understand what hsppened. Here is a sequence of comnands I did - not sure at wich point I removed octave itsekf, . Some commands failed but the autoremove finally worked as expected.

apt --fix-broken install
dpkg --remove --force-remove-reinstreq libopenblas
dpkg --remove --force-remove-reinstreq octave
apt remove octave
dpkg --remove --force-remove-reinstreq arpack-ng                                                  
  dpkg --remove --force-remove-reinstreq suitesparse
apt remove suitesparse
apt remove sundials
apt remove suitesparse
apt remove suitesparse
apt autoremove

**Comment by Richard092345** [comment]
I included the output of 'pkg install' because the output lists the subpackages that I couldn[t remove normally with the command 'pkg uninstall'. I think the output iof install is shorter than the output of uninstall. Some subpackages were not installed in the first place because 'pkg' refused to install those packages or any other package anyway. 'upgrade' did not work either. 'apt --fix-broken-install' attempted to instsll the missing subpackages but failed to do so.

As a result 'pkg' would neither install, remove or upgrade any package. At that point I filled in the bug report and looked deeper in the internals of 'apt' and I learned about the dpkg command which I used to unblock the normal operstion of the package manager.
